### PR TITLE
Avoid reflection by adding type hints

### PR DESCRIPTION
**Pull request by joda**
Makes it possible to use e.g. GraalVM

### CODE REVIEW

File: src/ring/middleware/basic_authentication.clj [hunk 1]
```diff
@@ -18,18 +18,18 @@
...
   (try
-    (s/join (map char (direction-fn (.getBytes string))))
+    (s/join (map char (direction-fn (.getBytes ^String string))))
     (catch Exception _)))
 
 (defn- encode-base64
   "Will do a base64 encoding of a string and return a string."
   [^String string]
-  (byte-transform #(.encode (Base64/getEncoder) %) string))
+  (byte-transform #(.encode ^java.util.Base64$Encoder (Base64/getEncoder) ^bytes %) string))
```
Comment (remvee) [inline src/ring/middleware/basic_authentication.clj:27]: Omitting `^java.util.Base64$Encoder` does not yield reflection warning.  Please remove when not needed.

File: src/ring/middleware/basic_authentication.clj [hunk 2]
```diff
@@ -18,18 +18,18 @@
...
 
 (defn- decode-base64
   "Will do a base64 decoding of a string and return a string."
   [^String string]
-  (byte-transform #(.decode (Base64/getDecoder) %) string))
+  (byte-transform #(.decode ^java.util.Base64$Decoder (Base64/getDecoder) ^bytes %) string))
```
Comment (remvee) [inline src/ring/middleware/basic_authentication.clj:32]: Same here.

File: project.clj [hunk 3]
```diff
@@ -10,4 +10,6 @@
 
   :codox {:output-path    "doc"
           :source-dir-uri ""
-          :source-uri     "https://github.com/remvee/ring-basic-authentication/blob/{git-commit}/{filepath}#L{line}"})
+          :source-uri     "https://github.com/remvee/ring-basic-authentication/blob/{git-commit}/{filepath}#L{line}"}
+
+  :global-vars {*warn-on-reflection* true})
```
Comment (remvee) [inline project.clj:15]: Not a big fan of global-var, please use `(set! *warn-on-reflection* true)` in the namespace.

**Review by remvee — CHANGES_REQUESTED** [review]
Thanks!  Please address comments.

**Comment by remvee** [comment]
Nevermind, I've cherry picked your commit and made the changes myself, see commit 0d20d01.

**Comment by remvee** [comment]
Oh, and pushed a new release (1.1.1).  Thanks for making this lib better!

**Comment by joda** [comment]
Thanks @remvee, got occupied with other stuff

**Comment by remvee** [comment]
No problem. I got impatient because some other issue popped up on this normally sleepy repo too. Thanks again.

**Pull request closed without merging**